src/BuildIntegration/Microsoft.NETCore.Native.Publish.targets

```diff
@@ -50,6 +50,9 @@
   <UsingTask TaskName="ComputeManagedAssemblies" AssemblyFile="$(IlcBuildTasksPath)" />
   <Target Name="_ComputeAssembliesToCompileToNative" DependsOnTargets="$(IlcDynamicBuildPropertyDependencies)">
 
+    <!-- Fail with descriptive error message for common mistake. -->
+    <Error Condition="'$(RuntimeIdentifier)' == ''" Text="RuntimeIdentifier is required for native compilation. Try dotnet publish -r option." />
```

Comment:
Could we be more explicit? i.e. "Try running dotnet publish with the -r option value specified"

Reply:
Fixed. Definitely looks better now.